### DISCUSSION TITLE
Make it component-based

### DISCUSSION THREAD

**Discussion by miladrahimi** [laravel/framework#40642] [Ideas]
Although the Laravel is getting bigger, it forces you to install the framework with all its features. I think it's time to split it into separate components.
For example, if you need to send emails in your project, you should first install the mail feature with the following command:
```compose require laravel/mail```
Or authentications:
```compose require laravel/session-auth```
```compose require laravel/sanctum```
And more important, let the users choose which one they need. Laravel should have no component installed out of the box. In the new version, you see sanctum as a separate package. It's great, but unfortunately, it has been installed out of the box.

**Comment by FoksVHox** [comment]
Couldn't you just aquire the specifics Illuminate packages you need? laravel/framework is just a large collection of https://github.com/illuminate

**Comment by miladrahimi** [comment]
@FoksVHox 
Laravel shouldn't install all of the components. It should let users decide which one to install and use. For example, if I need auth, I will install auth component.
There is another design problem with laravel components; They depend on each other. The HTTP Request class has user and validate methods and depends on them.
